org.polymc.PolyMC.yml

```diff
@@ -0,0 +1,130 @@
+app-id: org.polymc.PolyMC
+runtime: org.kde.Platform
+runtime-version: "5.15"
+sdk: org.kde.Sdk
+command: polymc
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=wayland
+  - --device=dri
+  - --share=network
+  - --socket=pulseaudio
+    # for Discord RPC mods
+  - --filesystem=xdg-run/app/com.discordapp.Discord:create
+
+modules:
+  # Compile PolyMC
+  - name: compile
+    buildsystem: simple
+    build-commands:
+      - mkdir build
+      - mkdir -p /app/bin
+      - cd build && JAVA_HOME=/run/build/compile/jdk PATH=$JAVA_HOME:$PATH JAVA_COMPILER=../jdk/bin/javac cmake -DLauncher_LAYOUT=lin-system -DCMAKE_INSTALL_PREFIX=/app/ .. && make -j$(nproc) install
+    sources:
+      - type: git
+        url: https://github.com/PolyMC/PolyMC.git
+        tag: 1.0.2
+      - type: archive
+        url: https://github.com/adoptium/temurin17-binaries/releases/download/jdk-17%2B35/OpenJDK17-jdk_x64_linux_hotspot_17_35.tar.gz
+        sha256: 6f1335d9a7855159f982dac557420397be9aa85f3f7bc84e111d25871c02c0c7
+        archive-type: tar-gzip
+        dest: jdk
+      - type: patch
+        path: patches/0001-Fix-DesktopServices.cpp-to-not-crash-when-using-flat.patch
+    cleanup: ["/share/applications/polymc.desktop"]
+  # old MC versions depend on Xrandr
+  - name: xrandr
+    buildsystem: autotools
+    sources:
+      - type: archive
+        url: https://xorg.freedesktop.org/archive/individual/app/xrandr-1.5.1.tar.xz
+        sha256: 7bc76daf9d72f8aff885efad04ce06b90488a1a169d118dea8a2b661832e8762
+    cleanup: ["/share/man", "/bin/xkeystone"]
+  # Install Java versions for different minecraft versions
+  - name: java
```

Comment:
There are also "official" OpenJDK extensions available on Flathub, but I don't know anything about Minecraft to say whether they are better suited for the job here. It was also used in the original MultiMC submission: https://github.com/flathub/flathub/pull/1978/files#diff-06d25db730a0affd31353a80acc5e64389d7ef7cad1898da8da7cd4000a23cdbR17

Reply:
I think this way is better.
- I think there are only LTS extensions, and some Minecraft versions depend on non-LTS javas (like minecraft 1.17 needs java 16)
- Only JRE is required for Minecraft, reducing the install size by a lot.

At the time of the official PR, I think minecraft only needed java 8